### PR TITLE
Setup log levels at environment

### LINKED ISSUE #30: Outputs to stdout gone when running maker
I just noticed now almost all the messages to stdout are missing when running the maker.

It seems the `log::trace!` is being made silent. Interestingly the taker messages are all outputted even though it also uses `trace!`

Presumably happened after #22

### PR DESCRIPTION
**Pull request by mojoX911**
Fixes #30 .

This should fix the logging issues.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -346,7 +347,23 @@ enum Subcommand {
...
+            .is_test(true)
+            .filter_module("teleport", log::LevelFilter::Trace)
+            .init();
+    });
+}
+
 fn main() -> Result<(), Box<dyn std::error::Error>> {
+    setup_logger();
+
+    log::trace!("Hello");
```
Comment (GeneFerneau) [inline src/main.rs:365]: nit: remove or replace with a more descriptive log startup message

**Comment by GeneFerneau** [comment]
Concept + code review ACK [2ba210d](https://github.com/bitcoin-teleport/teleport-transactions/pull/32/commits/2ba210dbc8311ee2759f1b72547fc5acfd0ac3d6)

File: src/main.rs (hunk 1 above)
Comment (mojoX911) [reply src/main.rs]: Ya this shouldn't be there. Removed.

**Comment by GeneFerneau** [comment]
reACK [a1c0a5f](https://github.com/bitcoin-teleport/teleport-transactions/pull/32/commits/a1c0a5f376816d3b405540efa2f6bb02b3821852)

**Comment by chris-belcher** [comment]
Please explain why the `Once::call_once` part is needed? main() and test_standard_coin_swap() will each only get invoked once I think?

**Comment by mojoX911** [comment]
> Please explain why the Once::call_once part is needed? main() and test_standard_coin_swap() will each only get invoked once I think?

`main()` will get evoked every time we say `teleport do-something`. So if we don't have the `Once` constrain, every time we run the binary the logging setup will try to initialise. This will cause error (in 2nd and subsequent calls of `teleport`) saying "`init()` should not be called once initialized". To mitigate that, this is the way to ensure we call `init()` only once, even though we run the binary multiple times.

**Comment by mojoX911** [comment]
Rebased and squashed.

**Comment by mojoX911** [comment]
Fixed the  bug. Updated logging initialization. 

Default logging is `info`. 

Can be changed to other levels by changing  this value
https://github.com/bitcoin-teleport/teleport-transactions/blob/330d56c63d790b9c2d1ea16085c8998857c66ed2/src/main.rs#L357

to filter specific object in debug log use `teleport=debug/<object-name>`.

Beyond that I have been playing around with log messages and it seems the info level messages can be refactored to show more readable protocol communication by default. This can be helpful for new devs trying to understand the communication flow.

Updated the `info` loggings to better reflect communications as they happen. While `debug` logs include all the object displays in pretty format.

This gives us log displays like below in `info` mode
```
[2021-09-03T17:29:27Z INFO  teleport::taker_protocol] ===> [Maker-0:6102] | Sending TakerHello
[2021-09-03T17:29:27Z INFO  teleport::taker_protocol] <=== [Internal:6102] | Received MakerHello
[2021-09-03T17:29:27Z INFO  teleport::taker_protocol] ===> [Maker-0:6102] | Sending Proof of Funding
[2021-09-03T17:29:31Z INFO  teleport::taker_protocol] <=== [Internal] | Recieved SignSendersAndReceiversContractTxes
[2021-09-03T17:29:31Z INFO  teleport::taker_protocol] [Maker-0:Self] | Taker is previous peer. Signing Receivers Contract Txs
[2021-09-03T17:29:31Z INFO  teleport::taker_protocol] ===> [0:16102] Maker-1 is next peer. Requesting to sign Sender's Contract Txs
```

The Taker logs are structured like `[<context>:<peer_port>] | <description>`.

Maker Log:
```
[2021-09-03T17:29:54Z INFO  teleport::maker_protocol] <=== [34058] | Accepted Connection From
[2021-09-03T17:29:54Z INFO  teleport::maker_protocol] ===> [34058] | Sending Maker Hello
[2021-09-03T17:29:54Z INFO  teleport::maker_protocol] <=== [34058] | Recieved TakerHello
[2021-09-03T17:29:54Z INFO  teleport::maker_protocol] <===[34058] | Recieved SignReceiversContractTx
[2021-09-03T17:29:54Z INFO  teleport::maker_protocol] ===> [34058] | Sending ReceiversContractSig
[2021-09-03T17:29:54Z INFO  teleport::maker_protocol] Reached EOF
```

Maker logs are simple `[peer_port] | <description>`.

I tried to cover all the message comms, so the flow can be seen.   

This unfortunately makes this PR little large. Happy to take it into a new PR if needed.

@chris-belcher @GeneFerneau  let me know what you think.

**Comment by mojoX911** [comment]
Updated with more clean logging.. That context thing for takers wasn't really helpful.. 

Now the logs are more concise:
```
$ ../target/debug/teleport --wallet-file-name=taker-wallet coinswap-send
[2021-09-10T18:17:56Z INFO  teleport::taker_protocol] <=== Got Offers
[2021-09-10T18:18:01Z INFO  teleport::taker_protocol] ===> [16102] | Sending SignSendersContractTx
[2021-09-10T18:18:01Z INFO  teleport::taker_protocol] <=== [16102] | Received SendersContractSig
[2021-09-10T18:18:01Z INFO  teleport::taker_protocol] Broadcasting My Funding Tx: 31b76f7f1a32277784b4e3d9b5bc4ded59f29825c26e36d901c2c07178e0e202
[2021-09-10T18:18:01Z INFO  teleport::taker_protocol] Broadcasting My Funding Tx: 504601b7ec5c2062c6cb66d88d8ea2b963abd49d93778d1c8714d8376c1936fe
[2021-09-10T18:18:01Z INFO  teleport::taker_protocol] Broadcasting My Funding Tx: 456bc03dfe355ebdc46129f0fee4d7c52901696cb5b2ede84507cab7f8a78b8b
[2021-09-10T18:18:01Z INFO  teleport::taker_protocol] Waiting for funding Tx to confirm
[2021-09-10T18:18:28Z INFO  teleport::taker_protocol] Funding Transaction confirmed
[2021-09-10T18:18:28Z INFO  teleport::taker_protocol] ===> [16102] | Sending ProofOfFunding
[2021-09-10T18:18:33Z INFO  teleport::taker_protocol] <=== [16102] | Recieved SignSendersAndReceiversContractTxes
[2021-09-10T18:18:33Z INFO  teleport::taker_protocol] Taker is previous peer. Signing Receivers Contract Txs
[2021-09-10T18:18:33Z INFO  teleport::taker_protocol] ===> [6102] | 6102 is next peer. Sending SignSendersContractTx
```

**Comment by chris-belcher** [comment]
I finally got round to trying this out. Thanks for the PR.

**Pull request merged**